### PR TITLE
Creates a basic mob version of bot mobs, and ports cleanbots to it

### PR DESCRIPTION
**Pull request by CapybaraExtravagante**
## About The Pull Request

I've been trying to get into basic mobs for a while now and after getting the hang of it I started porting bots to it. I started with the hygiene bot because it seemed simple enough while not being so simple that it didn't add enough core behavior.

The bots function the same (as much as possible anyways, because simple animals are a bit wonky). It shouldn't be too difficult to port the other bots over, so I'll hopefully be doing that after this PR :D

I made sure to check all uses of simple_animal/bot that it also accomodates for basic bots, and hopefully we can remove simple animal bots over time :) 

Some changes to datum AIs to accomodate this change:

- I added a set_movement_target method. This takes a second param that lets you assign a new movement datum. This was needed because JPS is used for patrolling, but the chasing uses basic avoidance!
- JPS movement had a hard-coded limit, this resulted in patrol points not working so I removed it and added a per-character limit
- Behaviors can now be set to not block planning; this is useful for behaviors that run without finishing. If the new plan does not contain this same behavior; it is cancelled.

and thanks to Armhulenn for all the help!!

## Why It's Good For The Game

Hopefully opens up the door to more interesting bot behavior in the future!

## Changelog

:cl: Capybara Holly
refactor: Cleanbots now use Datum AI, they should be slightly more responsive now!
/:cl:

### CODE REVIEW

**Review by MrMelbert — DISMISSED** [review]
`isbot()` should be updated with basic bots in mind

File: code/modules/modular_computers/file_system/programs/robocontrol.dm
Comment (CapybaraExtravagante) [reply code/modules/modular_computers/file_system/programs/robocontrol.dm:69]: ah thanks :) I need to check for occurances of simple_animal bot in for loops. I'll do a pass on that now!

**Comment by Mothblocks** [comment]
@tralezab

**Comment by tralezab** [comment]
I pogged! I'll be reviewing this for sure

**Comment by Mothblocks** [comment]
put yourself as codeowner for basics if possible arm

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/modular_computers/file_system/programs/robocontrol.dm
Comment (CapybaraExtravagante) [reply code/modules/modular_computers/file_system/programs/robocontrol.dm:139]: Yep, I also didn't like doing this. but seeing as we'll probably do it all in one go it should be ok,

File: code/modules/mob/living/basic/bots/bot_ai.dm
Comment (Mothblocks) [inline code/modules/mob/living/basic/bots/bot_ai.dm:235]: why the ocean

File: code/modules/mob/living/basic/bots/bot_ai.dm
Comment (tralezab) [inline code/modules/mob/living/basic/bots/bot_ai.dm:153]: more descriptive name than NB

File: code/modules/mob/living/basic/bots/bot_ai.dm
Comment (tralezab) [inline code/modules/mob/living/basic/bots/bot_ai.dm:195]: as MB says, get rid of some of this dead space

File: code/modules/mob/living/basic/bots/bot_ai.dm
Comment (tralezab) [inline code/modules/mob/living/basic/bots/bot_ai.dm:201]: dmdoc these

File: code/modules/mob/living/basic/bots/bot_ai.dm
Comment (tralezab) [reply code/modules/mob/living/basic/bots/bot_ai.dm:235]: remove it!

File: code/modules/mob/living/basic/bots/cleanbot/cleanbot_ai.dm
Comment (tralezab) [inline code/modules/mob/living/basic/bots/cleanbot/cleanbot_ai.dm:13]: unindent this line lel

File: code/modules/mob/living/basic/bots/hygienebot/hygienebot.dm
Comment (tralezab) [inline code/modules/mob/living/basic/bots/hygienebot/hygienebot.dm:95]: more descriptive name than A, please

File: code/modules/mob/living/basic/bots/hygienebot/hygienebot_ai.dm
Comment (tralezab) [inline code/modules/mob/living/basic/bots/hygienebot/hygienebot_ai.dm:13]: unindent this line

File: code/modules/mob/living/basic/bots/hygienebot/hygienebot_targetting
Comment (tralezab) [inline code/modules/mob/living/basic/bots/hygienebot/hygienebot_targetting:1]: this... isn't a .dm file?

File: code/modules/mob/living/silicon/ai/robot_control.dm
Comment (tralezab) [inline code/modules/mob/living/silicon/ai/robot_control.dm:93]: switch all these `.` uses in this proc to just return. Should be fine since actions are a switch case, and we're trying to minimize unnecessary use

File: code/modules/mob/living/simple_animal/bot/bot.dm
Comment (tralezab) [inline code/modules/mob/living/simple_animal/bot/bot.dm:521]: mistake?

**Review by tralezab — CHANGES_REQUESTED** [review]
mostly just cleaning up little bits of left over simple animal standards (aka lack of doc and single letter vars) and remembering to undo all the double blocks now that we're doing all the bots in this pr. stylewise you got it looking pretty spiffy

**Comment by Mothblocks** [comment]
@tralezab BTW don't merge this before my freeze test, then when that merges CapybaraExtravagante will need to pull from upstream and fix it (luckily very easy)

**Comment by Mothblocks** [comment]
Feel free to remove blocked after #70669

**Comment by Mothblocks** [comment]
Ok please merge from master

**Comment by CapybaraExtravagante** [comment]
getting all of the bots done will take some time :) I'll put the code into draft, but work continues!

File: code/modules/mob/living/basic/bots/bot.dm
Comment (CapybaraExtravagante) [reply code/modules/mob/living/basic/bots/bot.dm:372]: In what way shoul;d they be considered specifically? :)

**Comment by Mothblocks** [comment]
Now that master is merged in I'll remove blocked but you'll end up having to fix the test once you fix these compile errors, it's easy don't worry

**Comment by Mothblocks** [comment]
conflicts + merge from upstream

**Comment by Mothblocks** [comment]
This has been conflicting for 5 days, please let Armhulen know when you're ready.

In my opinion you should try to split this up into several PRs, it is much easier to review and we have a lot of huge refactors in our way right now

**Comment by CapybaraExtravagante** [comment]
thats what I wanted at first but it wasn't allowed, I had to do it all at once.

**Comment by CapybaraExtravagante** [comment]
I'm not really sure what to do with this PR but my motivation is kind of being sapped by the conflicting feedback T.T

**Comment by Mothblocks** [comment]
@CapybaraExtravagante What specifically wasn't allowed? I might be able to make something work

**Comment by CapybaraExtravagante** [comment]
I was told I couldn't port the bots one-by-one which made this PR bloat up a lot. It was all or nothing

**Comment by Mothblocks** [comment]
I definitely allow that if it makes the PR easier to review

**Pull request closed without merging**
